sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -113,48 +120,74 @@ def evaluate(
-        if i == 0 or strategy == "refit":
-            forecaster = forecaster.clone()
-            forecaster.fit(y_train, X_train, fh=fh)
-
-        else:  # if strategy in ["update", "no-update_params"]:
-            update_params = strategy == "update"
-            forecaster.update(y_train, X_train, update_params=update_params)
-        fit_time = time.perf_counter() - start_fit
-
-        pred_type = {
-            "pred_quantiles": "forecaster.predict_quantiles",
-            "pred_intervals": "forecaster.predict_interval",
-            "pred_proba": "forecaster.predict_proba",
-            None: "forecaster.predict",
-        }
-        # predict
-        start_pred = time.perf_counter()
-
-        if hasattr(scoring, "metric_args"):
-            metric_args = scoring.metric_args
-
         try:
-            scitype = scoring.get_tag("scitype:y_pred")
-        except ValueError:
-            # If no scitype exists then metric is not proba and no args needed
-            scitype = None
-            metric_args = {}
-
-        y_pred = eval(pred_type[scitype])(fh, X_test, **metric_args)
-
-        pred_time = time.perf_counter() - start_pred
-
-        # score
-        score = scoring(y_test, y_pred, y_train=y_train)
+            # fit/update
+            start_fit = time.perf_counter()
+            if i == 0 or strategy == "refit":
+                forecaster = forecaster.clone()
+                forecaster.fit(y_train, X_train, fh=fh)
+
+            else:  # if strategy in ["update", "no-update_params"]:
+                update_params = strategy == "update"
+                forecaster.update(y_train, X_train, update_params=update_params)
+            fit_time = time.perf_counter() - start_fit
+
+            pred_type = {
+                "pred_quantiles": "forecaster.predict_quantiles",
+                "pred_intervals": "forecaster.predict_interval",
+                "pred_proba": "forecaster.predict_proba",
+                None: "forecaster.predict",
+            }
+            # predict
+            start_pred = time.perf_counter()
+
+            if hasattr(scoring, "metric_args"):
+                metric_args = scoring.metric_args
+
+            try:
+                scitype = scoring.get_tag("scitype:y_pred")
+            except ValueError:
+                # If no scitype exists then metric is not proba and no args needed
+                scitype = None
+                metric_args = {}
+
+            y_pred = eval(pred_type[scitype])(fh, X_test, **metric_args)
+
+            pred_time = time.perf_counter() - start_pred
+
+            # score
+            score = scoring(y_test, y_pred, y_train=y_train)
+
+            # cutoff
+            cutoff = forecaster.cutoff
+
+        except Exception as e:
+            if error_score == "raise":
+                raise e
+            else:
+                warnings.warn(
+                    f"""
+                Fitting of forecaster failed, you can set error_score='raise' to see
```

Comment:
should this also print the exception message?

Reply:
I think no, bcz that will become a very long warning. user can still access exception message by setting `error_score="raise"`

Reply:
not easily if the `evaluate` is called from within elsewhere, e.g., grid search

Reply:
it will still raise the warning from `evaluate`